examples/animation/morph_targets.rs

```diff
@@ -0,0 +1,141 @@
+        ..default()
+    });
+    commands.spawn(DirectionalLightBundle {
+        directional_light: DirectionalLight {
+            color: Color::WHITE,
+            illuminance: 19350.0,
+            ..default()
+        },
+        transform: Transform::from_rotation(Quat::from_rotation_z(PI / 2.0)),
+        ..default()
+    });
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(3.0, 2.1, 10.2).looking_at(Vec3::ZERO, Vec3::Y),
+        ..default()
+    });
+}
+
+/// Marker for weights that are not animated by the animation system.
+#[derive(Component)]
+struct UpdateWeights;
+
+/// To update weights, query for [`MorphWeights`].
+///
+/// Note that direct children with a [`Handle<Mesh>`] component of entities
+/// with a [`MorphWeights`] component will inherit their parent's weights.
+fn update_weights(mut morphs: Query<&mut MorphWeights, With<UpdateWeights>>, time: Res<Time>) {
+    let mut t = time.elapsed_seconds();
+    let offset_per_weight = PI / 4.0;
+    for mut morph in &mut morphs {
+        for weight in morph.weights_mut() {
+            *weight = t.cos().abs();
+            t += offset_per_weight;
+        }
+    }
+}
+
+/// Deserialize the json field used in `gltf.mesh.extras` to associate
+/// weight indices to target names.
+#[derive(Component, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct TargetNames {
+    target_names: Vec<String>,
+}
+
+/// You can get the target names by reading the `GltfMeshExtras` component.
+/// They are in the order of the weights.
+fn name_morphs(query: Query<(&Name, &GltfMeshExtras)>, mut has_printed: Local<bool>) {
+    if *has_printed {
+        return;
+    }
+    for (name, extras) in &query {
+        info!("Node {name} has the following targets:");
+        let target_names: TargetNames = from_str(&extras.value).unwrap();
+        for name in &target_names.target_names {
+            info!("\t{name}");
+        }
+        *has_printed = true;
+    }
+}
+
+/// Read [`AnimationClip`]s from the loaded [`Gltf`] and assign them to the
+/// entities they control. [`AnimationClip`]s control specific entities, and
+/// trying to play them on an [`AnimationPlayer`] controlling a different
+/// entities will result in odd animations.
+fn setup_animations(
+    mut query: Query<
+        (&Name, Entity, Option<&mut AnimationPlayer>),
+        (With<MorphWeights>, Without<Handle<Mesh>>),
+    >,
+    gltf: Res<Assets<Gltf>>,
+    clips: Res<Assets<AnimationClip>>,
+    mut commands: Commands,
+    mut has_setup: Local<bool>,
+) {
+    if *has_setup {
+        return;
+    }
+    let Some((_, gltf)) = gltf.iter().next() else {
+        return;
+    };
+    // We check compatibility by getting the [`AnimationClip`] out of the
+    // [`Assets<AnimationClip>`] and using `compatible_with(&Name)`
+    let is_compatible = |name, clip| {
+        let Some(clip) = clips.get(clip) else { return false };
+        clip.compatible_with(name)
+    };
```

Comment:
This API is a bit involved. I think maybe the `Handle<AnimationClip>` could be stored in the `AnimationPlayer`, or a different component. But I didn't want to revamp the whole of `bevy_animation` because of #8357, so I decided to just provide this `compatible_with` method. to make it possible to use animation clips correctly with morph targets.

The issue already exists. Morph targets exacerbates it by causing a panic when they are misused. I just think it's not part of the scope of this already large PR.